.github/workflows/integration_tests.yml

```diff
@@ -49,189 +49,171 @@ env:
-        else
-          # Just a PR number.
-          echo "::set-output name=pr_number::${{ github.event.inputs.test_pull_request }}"
-          echo "::set-output name=github_ref::refs/pull/${{github.event.inputs.test_pull_request}}/merge"
-        fi
-    - name: Cancel previous runs on the same PR
-      uses: styfle/cancel-workflow-action@0.8.0
-      with:
-        access_token: ${{ github.token }}
-    - name: Remove triggering label
-      uses: actions-ecosystem/action-remove-labels@v1
-      with:
-        github_token: ${{ github.token }}
-        number: "${{ steps.get_pr_number.outputs.pr_number }}"
-        labels: "${{ github.event.label.name }}"
     ### Fail the workflow if the user does not have admin access to run the tests.
     - name: Check if user has permission to trigger tests
       uses: lannonbr/repo-permission-check-action@2.0.0
       with:
         permission: "admin"
       env:
         GITHUB_TOKEN: ${{ github.token }}
+    ### It sets "github_ref,trigger,pr_number,requested_tests" outputs to control the following jobs and steps
     - id: set_outputs
       run: |
-        echo "::set-output name=should_update_pr::1"
-        if [[ "${{ github.event.label.name }}" == "${{ env.triggerLabelFull }}" ]]; then
-          echo "::set-output name=requested_tests::full"
-        elif [[ "${{ github.event.label.name }}" == "${{ env.triggerLabelQuick }}" ]]; then
-          echo "::set-output name=requested_tests::auto"
-        else
-          # For manual PR testing, use auto mode.
-          echo "::set-output name=requested_tests::auto"
+        if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then 
+          if [[ "${{ github.event.inputs.test_pull_request }}" != "sdk" ]]; then
+            # Triggered manually
+            echo "::set-output name=trigger::manual_trigger"
+            if [[ "${{ github.event.inputs.use_expanded_matrix }}" == "1" ]]; then
+              echo "::set-output name=requested_tests::expanded"
+            fi
+            if [[ -z "${{github.event.inputs.test_pull_request}}" ]]; then
+              # test_pull_request not specified
+              echo "::set-output name=github_ref::$GITHUB_SHA"
+            elif [[ "${{github.event.inputs.test_pull_request}}" == *:* ]]; then
+              # If specified as pr:commit_hash, split them.
+              echo "::set-output name=github_ref::$(echo ${{ github.event.inputs.test_pull_request }} | cut -d: -f2)"
+              echo "::set-output name=pr_number::$(echo ${{ github.event.inputs.test_pull_request }} | cut -d: -f1)"
+            else
+              # Just the PR specified, use refs/pull/<number>/merge as the ref.
+              echo "::set-output name=github_ref::refs/pull/${{github.event.inputs.test_pull_request}}/merge"
+              echo "::set-output name=pr_number::${{ github.event.inputs.test_pull_request }}"
+            fi
+          elif [[ "${{ github.event.inputs.test_pull_request }}" == "sdk" ]]; then
```

Comment:
Remind me, did this value already exist? Why does it need to be set if we have the test_packaged_sdk input?

Reply:
This value already exist. https://github.com/firebase/firebase-cpp-sdk/blob/main/.github/workflows/cpp-packaging.yml#L725
Because it means it's from **scheduled** packaging workflow, thus it can generate nightly report. test_packaged_sdk only indicates that it's from packaging workflow.